app/main.cc

```diff
@@ -103,7 +103,7 @@ int main(int argc, char **argv)
       //try
       {
         std::string output_dir = wb_file.substr(0,wb_file.find_last_of("/\\") + 1);
-        world = std::make_unique<WorldBuilder::World>(wb_file, true, output_dir);
+        world = std::make_unique<WorldBuilder::World>(wb_file, INFINITY, true, output_dir);
```

Comment:
INFINITY is the old C stlye way of doing it. `std::numeric_limits<double>::infinity()` is the modern C++ way. The former is defined as a preprocessor `#define` as consequently does not actually have a type associated with it, just a bit pattern if you so want. The latter is unambiguously a `double` object.

It's a bit like `typedef Y X` vs `using X = Y`: The two more or less yield the same result, but `std::numeric_limits<double>::infinity()` is just more idiomatic C++ these days.

Reply:
I did https://github.com/GeodynamicWorldBuilder/WorldBuilder/pull/363

Reply:
Ah, I see. Yes, I agree that this is better, although I don't really like that it makes everything so long. I may make an alias of it at some point ;)

Thanks for making the pull request to fix this!

Reply:
Yes, I agree with the length :-( You can always do
```
namespace WorldBuilder
{
  namespace numbers {
    constexpr double infinity = std::numeric_limits<double>::infinity();
  }
}
```

An even better approach would probably be to not just use a "bare" infinity but give it a name that indicates what you actually want to express, namely an unbounded domain. Then you'd do
```
namespace WorldBuilder
{
  namespace numbers {
    constexpr double infinite_depth = std::numeric_limits<double>::infinity();
  }
}


void some_function (const double depth = infinite_depth);
```
and it is clear to the reader what the infinity in this case actually represents. In some sense, if you do this consistently, it doesn't even matter any more what specific value you give `infinite_depth`. It could have been `-1e12` or `NaN` or anything else because I assume you don't want to do arithmetic on the `INFINITY` anyway. All you want to do is compare a given `depth` variable against that "special value", and it no longer matters what value that actually is.

I'm a big fan of using variables for these kinds of "give special values a name" purposes.

Reply:
hmm, I like the idea of having constant values for specific functions. It reminds me a little bit of the type system in rust, which really encourages to use specialized types for different tasks (the enum system is really cool). On the other hand, it is probably not a good idea to have a ton of specialized values as global values laying around, but currently I don't thing we need that many. Although it could be set to any special value, setting it to infinity probably makes the most sense.

Reply:
In deal.II, we specifically use named values to indicate exceptional values: https://dealii.org/developer/doxygen/deal.II/namespacenumbers.html#a8ae36952c7e0cc778b47b5371b3aeff1 The point is that what the *numeric value* of these objects is doesn't matter, you just need to be able to compare against them for (in)equality. They're all typed with type names that indicate what they represent (see https://dealii.org/developer/doxygen/deal.II/namespacetypes.html). C++ doesn't allow defining integer types that cannot be converted into each other, but at least we can introduce a name like `types::boundary_id` that gives away what a variable is supposed to represent.